### PR TITLE
Working front-end update, and feedback loop wip first commits

### PR DESCRIPTION
**Pull request by rafaljanwojcik**
### Proposed Changes:
- modified `webapp.py` & `utils.py` in `ui` directory, to make front-end work for our use-case
-  started working on feedbacks modification, to make feedbacks loop work both for FAQ part, and Rulebook Extraction part - it's still a WIP, so it's not finished, and ideally I shouldn't have included that in this PR, but I started to work on it before making this PR, and I didn't want to revert the changes  I've already made to feedbacks, only to make this PR more readable - sorry for that, but everything in frontend apart from feedback loop is still working

### How did you test it?
Manual tests with Makefile (run 'make all' in terminal, and play with the Front-end tool)

### Notes for the reviewer
In terms of working frontend, only files: `Makefile`, `ui/utils.py`, and `ui/webapp.py` contain nessecary changes, other files are a WIP I've started when working on feedback loop

### CODE REVIEW

File: ui/utils.py [hunk 1]
```diff
@@ -68,10 +76,10 @@ def query(query, filters={}, top_k_reader=5, top_k_retriever=5) -> Tuple[List[Di
                 {
                     "context": "..." + answer["context"] + "...",
                     "answer": answer.get("answer", None),
-                    "source": answer["meta"]["name"],
+                    "source": answer["type"],
                     "relevance": round(answer["score"] * 100, 2),
                     "document": [doc for doc in response["documents"] if doc["id"] == answer["document_id"]][0],
-                    "offset_start_in_doc": answer["offsets_in_document"][0]["start"],
+                    # "offset_start_in_doc": answer["offsets_in_document"][0]["start"],
```
Comment (murielhol) [inline ui/utils.py:82]: maybe remove instead of comment out, can always trace back in git history

File: ui/webapp.py [hunk 2]
```diff
@@ -33,8 +39,8 @@ def set_state_if_absent(key, value):
 
 
 def main():
-
-    st.set_page_config(page_title="Haystack Demo", page_icon="https://haystack.deepset.ai/img/HaystackIcon.png")
+    icon_url = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSUp4mzekWcCkn3aSZQ2vrWB7Os8CYCtiMLlA&usqp=CAU"
+    st.set_page_config(page_title="Board Game Rules Explainer", page_icon=icon_url)
```
Comment (murielhol) [inline ui/webapp.py:43]: 👨‍🎨

**Review by murielhol — APPROVED** [review]
nice!! worked for me as well

File: ui/utils.py (hunk 1 above)
Comment (rafaljanwojcik) [reply ui/utils.py:82]: True, though I will remove it maybe in next PR, when I finish the feedback loop, to be 100% it won't be useful (if I remove it now, I will probably forget it even existed :D )

**Pull request merged**